canary/canary_ota.py

```diff
@@ -1,42 +1,44 @@
 import asyncio
 import time
+import datetime
 import copy
 import os
 
+import slack_sdk
+import slack_sdk.errors
+from dateutil import tz
+
 from viam.robot.client import DialOptions
 from viam.app.viam_client import ViamClient
 
+
 async def connect(robot_address: str, api_key: str, api_key_id: str) -> ViamClient:
-    dial_options = DialOptions.with_api_key(api_key_id=api_key_id,api_key=api_key)
-    return await ViamClient.create_from_dial_options(dial_options)
+    for i in range(5):
+        try:
+            dial_options = DialOptions.with_api_key(api_key_id=api_key_id, api_key=api_key)
+            viam_client = await ViamClient.create_from_dial_options(dial_options)
+            return viam_client
+        except Exception as e:
+            if i == 4:
+                raise e
+            print(e)
+        time.sleep(0.5)
 
 async def main():
-
     robot_address = os.environ["ESP32_CANARY_ROBOT"]
     api_key = os.environ["ESP32_CANARY_API_KEY"]
     api_key_id = os.environ["ESP32_CANARY_API_KEY_ID"]
     part_id = os.environ["ESP32_CANARY_ROBOT_PART_ID"]
     tag_name = os.environ["ESP32_CANARY_OTA_VERSION_TAG"]
-
+    bucket_url = os.environ["GCP_BUCKET_URL"]
+    bucket_name = os.environ["GCP_BUCKET_NAME"]
     bin_name = "micro-rdk-server-esp32-ota.bin"
-    url_base = "https://github.com/viamrobotics/micro-rdk/releases"
     
-    if tag_name == "latest":
-        url_target = f"{url_base}/latest/download/{bin_name}"
-    else:
-        url_target = f"{url_base}/download/{tag_name}/{bin_name}"
+    url = f"{bucket_url}/{bucket_name}/{tag_name}/{bin_name}"
```

Comment:
should the binary name also be an env variable?

Reply:
what else in our CI besides `micro-rdk-server-esp32-ota.bin` might go in this field?